Assignment/Assignment/PingProcess.cs

```diff
@@ -55,12 +71,29 @@ async public Task<PingResult> RunAsync(params string[] hostNameOrAddresses)
         return new PingResult(total, stringBuilder?.ToString());
     }
 
+    async public Task<PingResult> RunAsync(IEnumerable<string> hostNameOrAddresses, CancellationToken cancellationToken = default)
+    {
+        return await RunAsync(hostNameOrAddresses.ToArray());
+    }
+
     async public Task<PingResult> RunLongRunningAsync(
         string hostNameOrAddress, CancellationToken cancellationToken = default)
     {
-        Task task = null!;
-        await task;
-        throw new NotImplementedException();
+        StartInfo.Arguments = hostNameOrAddress;
+        StringBuilder stringBuilder = new();
+        void updateStdOutput(string? line) =>
+            (stringBuilder ??= new StringBuilder()).AppendLine(line);
+        Task<int> longRunning = RunLongRunningAsync(StartInfo, updateStdOutput, default, default);

```

Comment:
```suggestion
        Task<int> longRunning = RunLongRunningAsync(StartInfo, updateStdOutput, default, cancellationToken );

```
Make sure to use the cancellation token.